File: src/main/java/com/gitlab/daring/image/swing/BaseFrame.java
package com.gitlab.daring.image.swing;

import javax.swing.*;
import java.awt.*;

import static com.gitlab.daring.image.swing.SwingUtils.addWindowClosedListener;

public class BaseFrame extends JFrame {

    public BaseFrame(String title, Container content) {
        super(title);
        setContentPane(content);
        pack();
        setDefaultCloseOperation(DISPOSE_ON_CLOSE);
        setLocationByPlatform(true);
    }

    public void addCloseListener(Runnable l) {
        addWindowClosedListener(this, l);
    }

    public void show(int width, int height) {
        setSize(width, height);
        setVisible(true);
    }

}

File: src/main/kotlin/com/gitlab/daring/image/swing/BaseFrame.kt
package com.gitlab.daring.image.swing

import java.awt.Container
import java.awt.event.WindowAdapter
import java.awt.event.WindowEvent
import javax.swing.JFrame

class BaseFrame(title: String, content: Container) : JFrame(title) {

    init {
        contentPane = content
        pack()
        defaultCloseOperation = DISPOSE_ON_CLOSE
        isLocationByPlatform = true
    }

    fun addCloseListener(l: () -> Unit) {
        this.addWindowListener(object : WindowAdapter() {
            override fun windowClosed(e: WindowEvent) = l()
        })
    }

    fun show(width: Int, height: Int) {
        setSize(width, height)
        isVisible = true
    }

}